server/mailet/mailetcontainer-camel/src/main/java/org/apache/james/mailetcontainer/impl/JamesMailetContext.java

```diff
@@ -194,7 +197,10 @@ public void bounce(Mail mail, String message, MailAddress bouncer) throws Messag
 
         MailImpl reply = rawBounce(mail, message);
         // Change the sender...
-        reply.getMessage().setFrom(bouncer.toInternetAddress());
+        ConsumerChainer<InternetAddress> setFromConsumer = Throwing.consumer(reply.getMessage()::setFrom);
+        Optional.ofNullable(bouncer)
+            .map(MailAddress::toInternetAddress)
+            .ifPresent(setFromConsumer.sneakyThrow());
```

Comment:
really, you should not introduce this style when not necessary:
```
if (bouncer != null) {
  reply.getMessage.setFrom(bouncer.toInternetAddress())
}
```
is simpler no?

Reply:
:+1:

Reply:
done